File: proyectoRuby/lib/test_p1.rb
# encoding :UTF-8
require 'Dado'
require 'Enumerados'
require 'Mazo_Sorpresas'
require 'Sorpresa'
require 'Tablero'

module Civitas
  
  class TestP1
    def initialize
      main
    end
    
    def main
      
      # test quien empieza
      resultados = [0,0,0,0]
      for i in 0..99 do
        resultados[ Dado.instance.quien_empieza(4) ] += 1
      end
      
      puts ("Test quien empieza:")
      for i in 0..resultados.length-1 do
        puts ("Jugador " + (i+1).to_s + " empieza " + resultados[i].to_s + " veces")
      end
      puts ("\n")
      
      
      # test modo debug del dado
      Dado.instance.set_debug(true)
      for i in 0..2 do
        puts ("Tirada en modo debug, resultado: " + Dado.instance.tirar.to_s )
      end
      
      Dado.instance.set_debug(false)
      for i in 0..2 do
        puts ("Tirada sin modo debug, resultado: " + Dado.instance.tirar.to_s )
      end
      puts ("\n")
      
      
      # test ultimo_resultado y salgo_de_la_carcel
      puts ("El ultimo resultado fue: " + Dado.instance.ultimoResultado.to_s)
      puts ("Saldria de la carcel?: " + Dado.instance.salgo_de_la_carcel.to_s)
      puts ("\n")
      
      
      # test enumerados
      enum1 = Enumerados::TipoCasilla::CALLE
      enum2 = Enumerados::TipoSorpresa::PAGARCOBRAR
      enum3 = Enumerados::EstadosJuego::INICIO_TURNO
      puts ("enum1: " + enum1.to_s + "\nenum2: " + enum2.to_s + "\nenum3: " + enum3.to_s + "\n")
      
      
      # test mazo sorpresas
      mazo = MazoSorpresas.new
      sorpresa1 = Sorpresa.new("sorpresa1")
      sorpresa2 = Sorpresa.new("sorpresa2")
      mazo.al_mazo(sorpresa1)
      mazo.al_mazo(sorpresa2)
      
      puts ("la siguiente sorpresa es: " + mazo.siguiente.nombre)
      
      mazo.inhabilitar_carta_especial(sorpresa2)
      mazo.habilitar_carta_especial(sorpresa2)
      
      
      # test diario
      puts ("\nEventos pendientes en el diario:")
      while Diario.instance.eventos_pendientes do
        puts (Diario.instance.leer_evento)
      end
      
      # test tablero
      tablero = Tablero.new(5)
      tablero.añade_juez
  
      for i in 0..9 do
        tablero.añade_casilla(Casilla.new("casillatest"))
      end
   
      puts ("\nDespues de tirar los dados estoy en la casilla: " + tablero.nueva_posicion(0, Dado.instance.tirar).to_s)
    end
  end
  
  TestP1.new
end

File: proyectoJava/src/civitas/TestP1.java
package civitas;

public class TestP1 {
    public static void main(String[] argv) {
        
        // Test quienEmpieza()
        System.out.println("Test del método quienEmpieza():");
                
        int[] resultados = {0,0,0,0};
        
        for (int i = 0; i < 100; i++)
            resultados[ Dado.getInstance().quienEmpieza(4) ]++;
                        
        for (int i = 0; i < 4; i++)
            System.out.println("Jugador " + (i+1) + " empieza " + resultados[i] + " veces");
        
        
        // Test modo debug del dado
        System.out.println("\nTest del modo debug del dado:");
        Dado.getInstance().setDebug(true);
        for (int i = 0; i < 3; i++){
            int tirada = Dado.getInstance().tirar();
            System.out.println("Tirada con modo debug: " + tirada);
        };
        
        Dado.getInstance().setDebug (false);
        for (int i = 0; i < 3; i++){
            int tirada = Dado.getInstance().tirar();
            System.out.println("Tirada sin modo debug: " + tirada);
        };
        
        
        // Test métodos getUltimoResultado() y salgoDeLaCarcel()
        System.out.println("\nTest de los métodos getUltimoResultado() y salgoDeLaCarcel():");
        System.out.println("Último resultado: " + Dado.getInstance().getUltimoResultado());
        System.out.println("Sale de cárcel: " + Dado.getInstance().salgoDeLaCarcel());
        
        
        // Test tipos enumerados
        System.out.println("\nTest de los tipos enumerados:");
        TipoCasilla enum1 = TipoCasilla.CALLE;
        TipoSorpresa enum2 = TipoSorpresa.IRCARCEL;
        EstadosJuego enum3 = EstadosJuego.INICIO_TURNO;
        System.out.println(
                "Enumerado TipoCasilla: " + enum1 + "\n" +
                "Enumerado TipoSorpresa: " + enum2 + "\n" +
                "Enumerado EstadosJuego: " + enum3
        );
        
        
        // Test MazoSorpresas
        MazoSorpresas mazo = new MazoSorpresas();
        Sorpresa sorpresa1 = new Sorpresa();
        Sorpresa sorpresa2 = new Sorpresa();
        mazo.alMazo(sorpresa1);
        mazo.alMazo(sorpresa2);
        mazo.siguiente();
        mazo.inhabilitarCartaEspecial(sorpresa2);
        mazo.habilitarCartaEspecial(sorpresa2);
        
        // Test Diario
        Diario.getInstance().ocurreEvento("Test evento diario");
        Diario.getInstance().leerEvento();
        
        
        // Test tablero
        System.out.println("\nTest Tablero:");
        Tablero miTablero = new Tablero(5);
        miTablero.añadeJuez();
        for (int i = 0; i < 10; i++)
            miTablero.añadeCasilla(new Casilla("casilla_test"));
            
        System.out.println("Estoy en la casilla: " + miTablero.nuevaPosicion(0, Dado.getInstance().tirar() ) );
    }
}
